.travis.yml

```diff
@@ -1,40 +1,127 @@
-language: python
+conditions: v1
 
-sudo: false
+language: python
+python:
+- "3.7"
+- "2.7"
+- "pypy3.5-6.0.0"
+- "3.6"
+- "3.5"
+- "3.4"
+- "pypy2.7-6.0.0"
+- "2.6"
 
-cache: pip
+cache:
+  pip: true
+  directories:
+  - $HOME/.cache/pre-commit
+  - $HOME/.pre-commit
+  - $HOME/Library/Caches/Homebrew
 
-matrix:
-  include:
-    - python: 2.6
-      env: TOXENV=py26
-    - python: 2.7
-      env: TOXENV=py27
-    - python: 3.4
-      env: TOXENV=py34
-    - python: 3.5
-      env: TOXENV=py35
-    - python: 3.6
-      env: TOXENV=py36
-    - python: 3.7-dev
-      env: TOXENV=py37
-    - python: pypy
-      env: TOXENV=pypy
+install:
+- set -e
+- python -m pip.__main__ install cython tox
 
 # build libyaml
+- pushd /tmp
+- git clone https://github.com/yaml/libyaml.git -b "${LIBYAML_VERSION}" libyaml
+- cd libyaml
+- ./bootstrap
+- ./configure
+- make
+- make test-all
+- sudo make install
+- popd
+- set +e
+
 before_script:
-- >-
-  cd /tmp
-  && git clone https://github.com/yaml/libyaml.git -b 0.2.2-pre1 libyaml
-  && cd libyaml
-  && ./bootstrap
-  && ./configure
-  && make
-  && make test-all
-  && sudo make install
-  && sudo ldconfig
-  && cd "$TRAVIS_BUILD_DIR"
-
-install: pip install cython tox
-
-script: tox
+- sudo ldconfig
+
+script:
+- python -m tox.__main__
+
+.mixtures:
+- &osx_python
```

Comment:
_Remark:_ while this works, I've discovered a more flexible and faster way of doing this which I'm going to submit separately. The reference implementation is @ https://github.com/amol-/dukpy/blob/69f56f3/.travis.yml#L47-L120

Reply:
Implemented in this PR